src/PdfReader/generateSinglePdfToc.tsx

```diff
@@ -13,31 +9,46 @@ type PdfTocItem = {
 export default async function generateSinglePdfToc(
   proxiedUrl: string
 ): Promise<PdfTocItem[]> {
-  const tocItems: PdfTocItem[] = [];
   try {
     const pdf = await pdfjs.getDocument(proxiedUrl).promise;
     const outline = await pdf.getOutline(); // get the TOC outline
 
     if (outline) {
-      // iterate through each chapter
-      for (let i = 0; i < outline.length; i++) {
-        const dest = outline[i].dest;
-        if (dest && dest.length > 0) {
-          // get the page referance
-          const ref = dest[0];
-          const pageIndex = await pdf.getPageIndex(ref);
-          if (pageIndex) {
-            tocItems.push({
-              title: outline[i].title,
-              pageNumber: pageIndex + 1,
-            });
+      const addTocItem = async (
```

Comment:
Curious why this code is nested inside a function definition?

Reply:
There is no importable type for `PDFDocumentProxy`, so I nested the function to not need to pass `pdf` as type `any`.  I can update it to be a separate function definition and disable the type warning.